src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -1,22 +1,32 @@
-﻿using System.Net;
+using Orleans.Runtime;
+using System;
+using System.Net;
+using System.Net.Sockets;
 
 namespace Orleans.Hosting
 {
     /// <summary>
     /// Configures the Silo endpoint options
     /// </summary>
     public class EndpointOptions
-    { 
+    {
         /// <summary>
-        /// The external IP address or host name used for clustering.
+        /// The external IP address or host name used for clustering. IP address will be inferred from <see cref="NetworkAddress"/> if this is not
+        /// directly specified.
         /// </summary>
         public string HostNameOrIPAddress { get; set; }
```

Comment:
Could we maybe remove this property? Having a precedence order for properties is cumbersome/error prone.

We can have an async helper method for resolving an IPAddress from a string instead and let users consume that.